src/main/java/com/iota/iri/service/transactionpruning/jobs/MilestonePrunerJob.java

```diff
@@ -234,6 +235,7 @@ private void cleanupMilestoneTransactions() throws TransactionPruningException {
                             throw new RuntimeException(e);
                         }
                     }
+                    TransactionViewModel.cacheDelete(getTangle(), (Hash) element.low);
```

Comment:
This shows how hacky and bad the design is

It is of course not your fault... and it show how important it is to refactor the persistence layer (or DAL layer if we want to start speaking java)...

Anyhows, 
There is also `UnconfirmedSubtanglePrunerJob` that you missed


From here we learn that we need more regression/unit tests to be added...
Let's fix this and we can do this after the release

Reply:
oh, yeah.
We definitely need more regression tests to fish these out